rlp/codec.py

```diff
@@ -17,6 +15,29 @@
 from rlp.utils import ALL_BYTES
 
 
+def decode_raw(item, strict, preserve_per_item_rlp):
+    if preserve_per_item_rlp:
+        try:
+            # TODO: Maybe unify that in a common API on the rust side
+            return rusty_rlp.decode_raw_preserving(item, strict)
```

Comment:
The reason this is currently a separate API on the rust side is because preserving the inner rlps means allocating a bunch of extra Python arrays only for the use case of `recursive_cache` which is rarely used. Calling into Python code is a potential bottleneck and so I didn't want to make the happy case slower just to support a niche feature.

That said, I think we can still unify the API without ruining performance but I want to run benchmarks before I make any changes for that.

I just wanted to point out why we have this branching logic here.

Reply:
Ok, I have a [PR](https://github.com/cburgdorf/rusty-rlp/pull/10/files) coming that unifies the API without having a notable performance impact.

It then make it so, that we have such a single API:

```python
decoded, recursive_cache_info = rusty_rlp.decode_raw_preserving(item, strict=True, preserve_cache_info=True)
```

`recursive_cache_info` is the same array that currently is `per_item_rlp` but it will be an empty array if `preserve_cache_info` is set to `False`.